### PR TITLE
[qpid-proton] Fix #24280 and #24281 (PYTHON_EXECUTABLE and incorrect file path)

### LINKED ISSUE #24280: [qpid-proton] build failure
**Host Environment**
- OS: Windows
- Compiler: MSVC 16.11.8

**To Reproduce**
Add a dependency to `qpid-proton` in manifest mode

```json
        {
            "name": "qpid-proton",
            "version>=": "0.37.0"
        },
```

**Failure logs**
Inside the log file, there is

```shell
CMake Error at tools/cmake/Modules/FindPython.cmake:28 (message):
  Both PYTHON_EXECUTABLE and Python_EXECUTABLE are defined.  Define at most
  one of those.
Call Stack (most recent call first):
  C:/BuildTools/cmf/debug/vcpkg_installed/x64-windows/share/python/vcpkg-cmake-wrapper.cmake:67 (_find_package)
  C:/BuildTools/vcpkg/scripts/buildsystems/vcpkg.cmake:770 (include)
  CMakeLists.txt:45 (find_package)


-- Configuring incomplete, errors occurred!
```

**Additional context**
I'll open a PR to fix this in a few minutes.

### PR DESCRIPTION
**Pull request by klalumiere**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #24280 and #24281

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
 Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/qpid-proton/portfile.cmake [hunk 1]
```diff
@@ -22,16 +22,23 @@ vcpkg_cmake_configure(
         -DENABLE_BENCHMARKS=OFF
         -DENABLE_FUZZ_TESTING=OFF
         -DBUILD_TESTING=OFF
-        -DPYTHON_EXECUTABLE=${PYTHON3}
+        -DPython_EXECUTABLE=${PYTHON3}
```
Comment (klalumiere) [inline ports/qpid-proton/portfile.cmake:25]: This fixes #24280.

File: ports/qpid-proton/portfile.cmake [hunk 2]
```diff
@@ -22,16 +22,23 @@ vcpkg_cmake_configure(
...
 
 vcpkg_copy_pdbs()
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake)
+file(GLOB protonCmakeFiles LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/share/${PORT}/Proton/*.cmake")
+file(GLOB protonCppCmakeFiles LIST_DIRECTORIES false "${CURRENT_PACKAGES_DIR}/share/${PORT}/ProtonCpp/*.cmake")
+set(cmakeFiles ${protonCmakeFiles} ${protonCppCmakeFiles})
+foreach(cmakeFile IN LISTS cmakeFiles)
+    get_filename_component(filename "${cmakeFile}" NAME)
+    file(RENAME "${cmakeFile}" "${CURRENT_PACKAGES_DIR}/share/${PORT}/${filename}")
+endforeach()
```
Comment (klalumiere) [inline ports/qpid-proton/portfile.cmake:38]: This fixes #24281.

**Comment by Neumann-A** [comment]
duplication of #24267 ?

**Comment by klalumiere** [comment]
@Neumann-A 

Hmmm, #24267 will fix #24280, but not #24281. Hence, it is not a duplicate, but there will be merge conflict for sure. :slightly_frowning_face:

**Comment by LilyWangLL** [comment]
> @Neumann-A
> 
> Hmmm, #24267 will fix #24280, but not #24281. Hence, it is not a duplicate, but there will be merge conflict for sure. 🙁

@JackBoosY Could you please merge the fix for #24281 in this PR into your PR #24267？

**Comment by JackBoosY** [comment]
@LilyWangLL We'd better to merge #24267 first, then fix other issues in this PR.

**Comment by strega-nil-ms** [comment]
@klalumiere could you check and make sure your issues haven't been solved by #24267 ? I think now they should be.

**Comment by klalumiere** [comment]
@strega-nil-ms 

Super, it's all fixed, thanks! :tada:  It's interesting, I hadn't thought about using `vcpkg_cmake_config_fixup` [that way](https://github.com/microsoft/vcpkg/pull/24267/files#diff-f10f02582f4e7239437674b40d52a38d9a59e75301b2b4696a6681e9f1ea21a6R40-R48). :nerd_face:

**Pull request closed without merging**